From 040b7cdce0870f1e1e454aab9ee456cc06dcdcdd Mon Sep 17 00:00:00 2001
From: Robert Tiemann <rtie@gmx.de>
Date: Thu, 3 Sep 2015 13:24:53 +0200
Subject: [PATCH] mpegaudioparse: Fix delay samples computation.

In the previous version, an audible click could be heard when playing
sine wave files that are supposed to join seamlessly. Data written by
the multifilesink plugin revealed that the delay was one off by one
sample.
---
 gst/audioparsers/gstmpegaudioparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/audioparsers/gstmpegaudioparse.c b/gst/audioparsers/gstmpegaudioparse.c
index 83ec8f79b..0ada49c30 100644
--- a/gst/audioparsers/gstmpegaudioparse.c
+++ b/gst/audioparsers/gstmpegaudioparse.c
@@ -1169,7 +1169,7 @@ gst_mpeg_audio_parse_handle_first_frame (GstMpegAudioParse * mp3parse,
      * value the gap at the end of the segment. The decoder delay "shifts" these
      * two forward. As a result, the delay at the start is increased, but the
      * padding at the end is decreased. */
-    num_delay_samples = mp3parse->encoder_delay + decoder_delay;
+    num_delay_samples = mp3parse->encoder_delay + decoder_delay + 1;
     if (mp3parse->encoder_padding < decoder_delay)
       num_padding_samples = 0;
     else
-- 
2.17.1

